dask/dataframe/io/hdf.py

```diff
@@ -473,6 +473,8 @@ def read_hdf(
         paths = sorted(glob(pattern))
     else:
         paths = pattern
+    if len(paths) == 0:
+        raise FileNotFoundError("No matching file(s) found for path")
```

Comment:
Can you include the user-provided path here? Can you also ensure that we have a test for this?

And is this the correct behavior for `dd.read_hdf([])`? I'm not sure what's best in that case...

Reply:
> Can you include the user-provided path here? Can you also ensure that we have a test for this?

Yes, I'll add both.

> And is this the correct behavior for ``dd.read_hdf([])``? I'm not sure what's best in that case...

I'll look into how other read functions behave when passed an empty list, and find out what is the proper return value or Exception.

Reply:
- [ ] Include original path in Exception
- [ ] Add tests
- [ ] Find out what to do for empty lists